language-design.md

```diff
@@ -17,7 +17,81 @@ For each quote, describe what it was about the quote that led you pick it.
 
 **Response**
 
-
+***Finding the Balance in Creation: Cathedral versus Bazaar***
+
+> The key point is that in the bazaar style of building a program or designing a
+> language or what you will, the plan can change in real time to meet the needs
+> of those who work on it and use it. This tends to make users stay with it as
+> time goes by; they will take joy in working hard and helping out if they know
+> that their wants and needs have some weight and their hard work can change the
+> plan for the better [Steele, 1998]
+
+While flexibility as emphasized in this quote and throughout Steele's paper
+are foundational to good software development and general-purpose programming
+language development, we feel that a stricter approach when building out a DSL
+is essential in avoiding feature-bloat. Anthony pointed out an analagous duality
+of development styles prompted by this quote: Cathedral-Style is to Bazaar-Style
```

Comment:
This is a really good analogy! I didn't know about the Waterfall and Agile models before, but it relates nicely with the dichotomy of the Cathedral vs. Bazaar in the language design process. I also agree that a DSL might require a more Cathedral-style approach because by definition, the scope of a DSL needs to be more limited. At the same time, designing a language, whether a DSL or a GPL, will always require a degree of balance between the two approaches, and I think this is the point that Steele makes when he argues for a "Shopping Mall" approach, where there aren't too many choices for the user but the choices are fully fleshed out and well-designed.